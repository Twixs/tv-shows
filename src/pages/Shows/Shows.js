import React, { useEffect } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { useParams } from 'react-router-dom';
import { getCurrentShowDataStart, selectCurrentShow } from '../../store/slices';
import s from './Shows.module.scss';
import { InfoBlock } from '../../components/InfoBlock/InfoBlock';
import { EpisodeListItem } from './components/EpisodeListItem/EpisodeListItem';

export const Shows = () => {
  const { id } = useParams();
  const dispatch = useDispatch();
  const { show, episodes, loading } = useSelector(selectCurrentShow);

  useEffect(() => {
    dispatch(getCurrentShowDataStart(id));
  }, [dispatch, id]);

  if (loading || !show) {
    return <p>Loading...</p>;
  }

  return (
    <div className={s.ShowWrapper}>
      <InfoBlock data={show} />
      {episodes && (
        <>
          <h3>Episodes</h3>
          {episodes.map((episode) => (
            <EpisodeListItem key={episode.id} episode={episode} />
          ))}
        </>
      )}
    </div>
  );
};
