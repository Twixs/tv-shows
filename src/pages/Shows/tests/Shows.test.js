import React from 'react';
import renderer from 'react-test-renderer';
import configureMockStore from 'redux-mock-store';
import { createMemoryHistory } from 'history';
import { Router } from 'react-router-dom';
import { Provider } from 'react-redux';
import { getByText, render } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import { Shows } from '../Shows';

const loadingState = {
  loading: true,
  show: null,
  episodes: null,
  error: false,
};

const showInfoState = {
  loading: false,
  show: {
    id: 1,
    name: 'My test show',
    summary: '<p>Testing test</p>',
  },
  episodes: [
    {
      id: 1,
      name: 'My test episode',
      summary: '<p>Testing test episode</p>',
    },
  ],
  error: false,
};

const mockStore = configureMockStore();

describe('Shows Page', () => {
  const history = createMemoryHistory();

  it('has loading message', () => {
    const store = mockStore({ currentShow: loadingState });
    const { getByText } = render(
      <Router history={history}>
        <Provider store={store}>
          <Shows />
        </Provider>
      </Router>
    );

    expect(getByText(/loading\.\.\./i)).toBeInTheDocument();
  });

  it('renders correctly', () => {
    const store = mockStore({ currentShow: showInfoState });
    const tree = renderer
      .create(
        <Router history={history}>
          <Provider store={store}>
            <Shows />
          </Provider>
        </Router>
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
