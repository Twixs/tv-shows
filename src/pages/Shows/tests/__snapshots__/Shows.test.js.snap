// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Shows Page renders correctly 1`] = `
<div
  className="ShowWrapper"
>
  <div
    className="InfoWrapper"
  >
    <img
      alt="My test show"
    />
    <div
      className="Details"
    >
      <h2>
        My test show
      </h2>
      <p
        dangerouslySetInnerHTML={
          Object {
            "__html": "<p>Testing test</p>",
          }
        }
      />
    </div>
  </div>
  <h3>
    Episodes
  </h3>
  <div
    className="ListItem"
  >
    <p
      className="ListItemShortInfo"
    >
      x
      :
    </p>
    <a
      href="/episodes/1"
      onClick={[Function]}
    >
      My test episode
    </a>
  </div>
</div>
`;
